app/domain/filter_params.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class FilterParams
+  def initialize(params)
+    @params = params
+  end
+
+  def filters_and_results
+    filters = search_filters
+    results = search_results(filters.filter { |f| f.exclude?(nil) })
+    [filters, results]
+  end
+
+  private
+
+  def search_filters
+    unless skill_ids && levels && interests
+      return [default_filter]
+    end
+
+    filters = constructed_filters
+    filters << default_filter if add_row?
+    filters.delete_at(delete_row.to_i) if delete_row
+
+    filters
+  end
+
+  def search_results(filters)
+    return [] unless filters.any?
+
+    skill_ids = filters.transpose.first
```

Comment:
Have you considered using a hash instead so you could used named keys instead of having to use `.first` ?